lib/modules/platform/gitlab/index.ts

```diff
@@ -757,6 +757,44 @@ export async function createPr({
   }
   const description = sanitize(rawDescription);
   logger.debug(`Creating Merge Request: ${title}`);
+
+  let reviewerIds: number[] = [];
+  if (platformPrOptions?.gitLabReviewersFromApprovalRule) {
+    logger.debug(
+      `Fetching reviewers from GitLab approval rule: ${platformPrOptions.gitLabReviewersFromApprovalRule}`,
+    );
+    try {
+      const approvalRules = await gitlabApi.getJson(
+        `projects/${config.repository}/approval_rules`,
+        GitLabApprovalRules,
+      );
+
+      const approvalRuleReviewerIds = new Set<number>();
+      const matchingApprovalRule = approvalRules.body.find(
+        (rule) =>
+          rule.name === platformPrOptions.gitLabReviewersFromApprovalRule,
+      );
+
+      if (matchingApprovalRule?.eligible_approvers?.length) {
+        logger.debug('Found matching approval rule');
+        for (const approver of matchingApprovalRule.eligible_approvers) {
+          approvalRuleReviewerIds.add(approver.id);
+        }
+        reviewerIds = Array.from(approvalRuleReviewerIds);
+      } else {
+        logger.debug(
+          'No matching approval rule found or rule has no eligible approvers',
+        );
+      }
+      logger.debug(
```

Comment:
Move this inside the "if" above so that it's only logged if the array is non empty